src/layout/Grid.scss

```diff
@@ -0,0 +1,86 @@
+.grid {
+  --grid-gap-sm: var(--bloom-s4);
+  --grid-gap-md: var(--bloom-s6);
+  --grid-gap-lg: var(--bloom-s8);
+  --grid-gap-xl: var(--bloom-s12);
+
+  --grid-gap-tablet: var(--grid-gap-sm);
+
+  --grid-gap: var(--grid-gap-md);
+  
+  @media (max-width: 1023px) {
+    --grid-gap: var(--grid-gap-tablet);
+  }
+
+  display: grid;
+  gap: var(--grid-gap);
+
+  @media (min-width: 1024px) {
+    &[data-spacing="sm"] {
+      --grid-gap: var(--grid-gap-sm);
+    }
+
+    &[data-spacing="md"] {
+      --grid-gap: var(--grid-gap-md);
+    }
+
+    &[data-spacing="lg"] {
+      --grid-gap: var(--grid-gap-lg);
+    }
+
+    &[data-spacing="xl"] {
+      --grid-gap: var(--grid-gap-xl);
+    }
+  }
+}
+
+.grid-row {
+  --grid-columns: auto-fit;
+  --grid-minimum-column-size: 0;
+
+  @media (max-width: 639px) {
+    --grid-columns: 1;
+  }
+
+  display: grid;
+  gap: inherit;
+  grid-template-columns: repeat(var(--grid-columns), minmax(var(--grid-minimum-column-size), 1fr));
+}
+
+.grid-row > * {
+  overflow-wrap: break-word;
+}
+
+@media (min-width: 640px) {
+  .grid-row[data-columns="4"] {
+    --grid-columns: 4;
+  }
+
+  .grid-row[data-columns="3"] {
+    --grid-columns: 3;
+  }
+
+  .grid-row[data-columns="2"] {
+    --grid-columns: 2;
+  }
+
+  .grid-row[data-columns="1"] {
+    --grid-columns: 1;
+  }
```

Comment:
I haven't been in all of the discussions so there might be some scenario that I'm missing so feel free to ignore this comment if that is the case.

I'm curious what the use case of having to explicitly set the grid column count beyond the "1+2" or the "2+1"? It seems to me like if you want three columns you would just put three cells in the row and then any overflow would go into a new row?

Reply:
@jaredcwhite any insight into this?